### ISSUE TITLE
Disable `no-curly-component-invocation` rule in `octane` configuration

### ISSUE THREAD

**Issue by Turbo87** [ember-template-lint/ember-template-lint#925]
The `no-curly-component-invocation` rule in the `octane` configuration is creating a lot of false positives as it has no knowledge of what is a helper or component and in some cases it also flags regular `{{foo}}` values as components.

To avoid this unnecessary noise for Octane users we should disable the rule until we figure these issues out.

The second issue is clearly a bug, the first one can't really be solved, but we could still use a better heuristic and e.g. consider curlies with positional args as helpers.

/cc @rwjblue @lifeart

**Comment by lifeart** [comment]
@Turbo87  try define `ember-template-lint` dependency for app, (with ember-cli-template-lint)
ember-cli-template-lint has old version, after adding "upsteam" `ember-template-lint` into dev-devs, issue fixed

![image](https://user-images.githubusercontent.com/1360552/68881749-c5f80300-071e-11ea-91b7-fd7c72cacd42.png)

![image](https://user-images.githubusercontent.com/1360552/68881773-d4461f00-071e-11ea-918e-e248a1660240.png)

**Comment by lifeart** [comment]
@rwjblue @Turbo87 I think we need new `ember-cli-template-lint` version

**Comment by lifeart** [comment]
more context:

![image](https://user-images.githubusercontent.com/1360552/68883962-2be68980-0723-11ea-8e8e-1951044e22c8.png)

**Comment by lifeart** [comment]
In theory, we can extract lookup part of lookup logic from `ember-language-server`, and load all helpers. (including deps) :)

**Comment by patocallaghan** [comment]
@Turbo87 For differentiating between helpers and components you need to whitelist your helpers. The rule will throw on any curly expression by default https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-curly-component-invocation.md#whitelisting-helpers

But I agree that the out-of-the-box experience is a bit 😱

**Comment by rwjblue** [comment]
> The no-curly-component-invocation rule in the octane configuration is creating a lot of false positives as it has no knowledge of what is a helper or component and in some cases it also flags regular {{foo}} values as components.

Can we discuss these false positives a bit more? Are you primarily seeing helpers that are invoked and appended to the DOM directly (IMHO these are _fairly_ rare)? Something else?

At this point, I'd prefer to focus on reducing false positives as much as possible, I'm not opposed to disabling for the octane preset if we can't make it an acceptable experience.

**Comment by Turbo87** [comment]
some examples of false positives:

```hbs
{{svg-jar "status"}}
```

```hbs
{{t "some.translation.key"}}
```

```hbs
{{model.selectedTransfersCount}}
```

```hbs
{{request.note}}
```

as I mentioned above, I would consider the second two cases bugs. they have no `this.` or `@`, but I think we still shouldn't be flagging every single curly as a component by default 🤔 

the first two cases are common helpers from `ember-svg-jar` and `ember-intl`, we could adjust our whitelist, but I'd prefer to use the mentioned heuristic by default, where a curly that receives a positional argument is considered a helper invocation instead of a component.

**Comment by rwjblue** [comment]
> a curly that receives a positional argument is considered a helper invocation instead of a component

I think this is fine, but it isn't that we are assuming it is or isn't a helper. It's more that components (or helpers!) that accept positional params can't be converted to angle bracket invocation anyways.

> ```
> {{model.selectedTransfersCount}}
> ```
> 
> 
> ```
> {{request.note}}
> ```

I think these two scenarios can be finessed. Specifically, we could treat ambiguous non-block mustache statements (those without arguments) as property lookups and not assume they are component invocations.

---

It seems likely that given these two fixes, that we would be fine:

* do not flag invocations with positional params
* do not flag ambiguous mustaches as component invocations

**Comment by Turbo87** [comment]
yep, that would be fine by me. I think it's fine to have a strict mode where you explicitly specify your helpers etc., but the default should avoid most unnecessary noise IMHO.

**Comment by rwjblue** [comment]
> the default should avoid most unnecessary noise

Yep, I totally agree, this is my stance as well. We just need bug reports when we aren't hitting that mark. :smile:

**Comment by patocallaghan** [comment]
So what's the consensus here? To make the two fixes above and have an opt-in strict mode? I'd be happy to do the work if so.

1. Do not flag invocations with positional params (straight forward to understand)
2. Do not flag ambiguous mustaches as component invocations (in this context what does ambiguous look like?)
3. Introduce a strict mode. Is this a config option? And does opting into it disable the above two fixes? 

I guess we haven't encountered these issues at work because we enabled the `requireDash: true` option whereas in Octane preset the default is `false`.

**Comment by rwjblue** [comment]
@patocallaghan - Thanks for offering to work on it! 

I think these two are "ready" for implementation work:

> Do not flag invocations with positional params (straight forward to understand)
> Do not flag ambiguous mustaches as component invocations

Then we can see where things settle and decide if we need another config option for strict (it's still not 100% clear what this would mean, so fixing the first two would help clarify).

> in this context what does ambiguous look like?

Ambiguous specifically means:

A `MustacheStatement` (does not apply to `BlockStatements` as they are never ambiguous), with no positional arguments and no named arguments. They would look like:

```
{{foo}}
```

**Comment by patocallaghan** [comment]
@rwjblue I've finally got some time to work on these fixes. I just have a clarification question.

We previously shipped a fix in https://github.com/ember-template-lint/ember-template-lint/pull/861 which implemented the following logic:

> We could tweak things a bit so that the rule does not apply to block params when used without any arguments (e.g. {{foo}} vs either {{foo whatever}} or {{foo some=thing}}).

So this slightly contradicts the feedback above (or maybe it doesn't 🤔)

> Do not flag invocations with positional params (straight forward to understand)

Does the change I'm making here, `Do not flag invocations with positional params` apply to be block params and non-block params, or just to non-block params? Do I need to refactor the previous change we shipped?
